internal/account/service.go

```diff
@@ -13,13 +14,19 @@ import (
 
 type accountDBAccessor interface {
 	RegisterAccount(ctx context.Context, account Account) error
+	FindAccountByEmail(ctx context.Context, email string) (*Account, error)
+}
+
+type tokenService interface {
+	GenerateToken(spec token.ClaimSpec) (string, error)
 }
 
 type AccountService struct {
 	accountDBAccessor
+	tokenService
 }
 
-func (a *AccountService) RegisterAccount(ctx context.Context, spec RegisterAccountSpec) error {
```

Comment:
Changing this might affect another branch. Make sure that it will not cause any conflict